pkg/util/httputil/http.go

```diff
@@ -4,24 +4,42 @@
 package httputil
 
 import (
+	"crypto/tls"
 	"fmt"
 	"io"
 	"net/http"
+	"time"
 
 	"github.com/cenkalti/backoff/v4"
 	"github.com/pingcap/tiproxy/lib/util/errors"
 	pnet "github.com/pingcap/tiproxy/pkg/proxy/net"
 )
 
-func Get(httpCli http.Client, addr, path string, b backoff.BackOff) ([]byte, error) {
+type Client struct {
+	cli          *http.Client
+	getTLSConfig func() *tls.Config
+}
+
+func NewHTTPClient(getTLSConfig func() *tls.Config) *Client {
+	return &Client{
+		cli: &http.Client{
+			Transport: &http.Transport{TLSClientConfig: getTLSConfig()},
+		},
+		getTLSConfig: getTLSConfig,
+	}
+}
+
+func Get(c Client, addr, path string, b backoff.BackOff, timeout time.Duration) ([]byte, error) {
+	httpCli := NewHTTPClient(c.getTLSConfig)
```

Comment:
I mean, the server shares the same `httpCli` but copies `cli`. The `cli` is shared and `Timeout` is written concurrently.